### ISSUE TITLE
Cant hide x-axis on API generated timeseries panels

### ISSUE THREAD

**Issue by bobemoe** [grafana/grafana#65001]
**What happened**:
I generate some dashboards by pushing up JSON through the grafana API. I start by designing the panel in the web interface as usual, copying the json to my application and then using that panel on various auto generated dashboards. But I notice that the x-axis labels that were hidden, are now showing once pushed up. In a small panel this completely obscures the graph. See before and after screenshot below:

![image](https://user-images.githubusercontent.com/2304970/226169529-7849c362-1371-434b-8dc3-69f69e9d39a9.png)

**What you expected to happen**:

Panel with identical JSON should look the same.

**How to reproduce it (as minimally and precisely as possible)**:

Create a new panel, add a random walk or simple test series. Set axis placement to hidden. Observe both axis hide. Copy panel/dashboard JSON, push the JSON up to the dashboard using the API. See now the x-axis labels have appeared.

**Anything else we need to know?**:

I have done a diff on the original JSON vs that of the panel once pushed up through the API. It appears this is added to the `fieldConfig`->`overrides` section:

```
      {
        "matcher": {
          "id": "byType",
          "options": "time"
        },
        "properties": [
          {
            "id": "custom.axisPlacement",
            "value": "auto"
          }
        ]
      }
```

Changing auto to hidden does hide the labels, so thought I could therefore add that to the source JSON, but when I tried it was still added and I ended up with:

```
      {
        "matcher": {
          "id": "byType",
          "options": "time"
        },
        "properties": [
          {
            "id": "custom.axisPlacement",
            "value": "hidden"
          }
        ]
      },
      {
        "matcher": {
          "id": "byType",
          "options": "time"
        },
        "properties": [
          {
            "id": "custom.axisPlacement",
            "value": "auto"
          }
        ]
      }
```

**Environment**:
- Grafana version: 9.4.3
- Data source type & version: Grafana
- OS Grafana is installed on: Docker
- User OS & Browser: Firefox Linux
- Grafana plugins: 
- Others:

**Comment by zuchka** [comment]
could you please copy and paste all relevant data using the `panel --> more --> get help` workflow in the panel's dropdown menu? This will help up mock up your data and reproduce your issue. It is important to our teams that issues get verified as reproducible before we formally mark them as bugs. Thank you!

here are more details and steps for copying your data over to this issue:

https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/

![CleanShot 2023-01-31 at 14 40 21](https://user-images.githubusercontent.com/37156449/215916832-8d29d39c-57a1-40a7-a6e0-bfe584ea8bd4.gif)

**Comment by bobemoe** [comment]
Ok, so I've figured this out now...

If I pull then push the whole dashboard through the API then it does not get altered.  Its only when I push up a custom generated dashboard that the problem occurred.

Yes I am using CURL (php and bash), based on the information found here: https://grafana.com/docs/grafana/latest/developers/http_api/dashboard/ and when I add the `panels[]` section copied from the working dashboard to that example and push that up then I see the issue.

I noticed while pulling the working dashboard that `schemaVersion` is 36.  In the API example above that I based my code on its version 16. I changed the version in my json from 16 to 36 and now the problem disappears!

Not sure exactly the implications of changing this? If its safe to do so maybe the example API docs need to be updated?

@zuchka Happy for you to close this now, but can still provide the further info if it would be useful?
